bundle/sign.go

```diff
@@ -7,16 +7,30 @@ package bundle
 
 import (
 	"encoding/json"
+	"fmt"
 
 	"github.com/open-policy-agent/opa/internal/jwx/jwa"
 	"github.com/open-policy-agent/opa/internal/jwx/jws"
 )
 
+const defaultSignerID = "_default"
+
+var signers map[string]Signer
+
+// Signer is the interface expected for implementations that generate bundle signatures.
+type Signer interface {
+	GenerateSignedToken([]FileInfo, *SigningConfig, string) (string, error)
+}
+
+// DefaultSigner is the default bundle signing implementation. It signs bundles by generating
+// a JWT and signing it using a locally-accessible private key.
+type DefaultSigner struct{}
+
 // GenerateSignedToken generates a signed token given the list of files to be
 // included in the payload and the bundle signing config. The keyID if non-empty,
 // represents the value for the "keyid" claim in the token
-func GenerateSignedToken(files []FileInfo, sc *SigningConfig, keyID string) (string, error) {
-	payload, err := generatePayload(files, sc, keyID)
+func (*DefaultSigner) GenerateSignedToken(files []FileInfo, sc *SigningConfig, keyID string) (string, error) {
+	payload, err := GeneratePayload(files, sc, keyID)
```

Comment:
Do we need to export `GeneratePayload` ?

Reply:
yea, because the payload generated doesn't really change between signers, at least for our use case.  so having access to that in our custom signer removes the need to duplicate the code.

Reply:
I think let's not export this for now. It's possible that custom signers have their own way of generating the payload. The function itself is pretty small so the code duplication is not much. We can always export it in the future if needed.